packages/drivers/odsp-driver/src/retryUtils.ts

```diff
@@ -44,7 +44,7 @@ export async function runWithRetry<T>(
             const coherencyError = error?.[Odsp409Error] === true;
             const serviceReadonlyError = error?.errorType === OdspErrorType.serviceReadOnly;
             // Retry for retriable 409 coherency errors or serviceReadOnly errors.
-            if (!(coherencyError || serviceReadonlyError || canRetry)) {
+            if (!((coherencyError || serviceReadonlyError) && canRetry)) {
```

Comment:
I'd add these errors are retryable, but some code can patch canRetry to make whole flow non-retryable, and thus we should bail out.